tests/extmod/socket_fileno.py

```diff
@@ -0,0 +1,17 @@
+# Test socket.fileno() functionality
+
+try:
+    import socket, errno
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+if not hasattr(socket.socket, "fileno"):
+    print("SKIP")
+    raise SystemExit
+
+s = socket.socket()
+assert s.fileno() >= 0
```

Comment:
```suggestion
print(s.fileno() >= 0)
```

In Micropython tests, we usually print something to compare to CPython rather than doing an `assert`.